schema.sql

```diff
@@ -1,11 +1,50 @@
 CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
 
+CREATE TABLE courses (
+  course_id serial PRIMARY KEY,
+  course_code varchar(5) NOT NULL,
+  course_name text
+);
+
+CREATE TABLE tracks (
+  track_id serial PRIMARY KEY,
+  track_code varchar(2) NOT NULL,
+  track_name text
+);
+
+CREATE TABLE timezones (
+  tzone_id serial PRIMARY KEY,
+  tzone_code varchar(5) NOT NULL,          -- e.g EST, CET, MIT,
+  tzone_name text,                         -- e.g Eastern Standard Time, Central European Time, Marquesas Islands Time
+  gmtoffset numeric(4,2) NOT NULL          -- e.g -5, 1, -9.50
+);
+
+
 CREATE TABLE users (
-  pkey UUID NOT NULL DEFAULT uuid_generate_v1(),
-  firstname text NOT NULL CHECK (LENGTH(TRIM(FROM firstname)) > 0),
+  user_id serial, -- change datatype from serial to uuid DEFAULT uuid_generate_v4 ()
+  firstname text NOT NULL CHECK (LENGTH(TRIM(BOTH ' ' FROM firstname)) > 0), -- should we use the checks in the form instead of here?
   lastname text,
-  -- TODO: ADD REMAINING COLUMNS
-  created_at timestamp DEFAULT NOW()
+  slackname text NOT NULL,
+  email varchar(255) NOT NULL,
+  password text NOT NULL,
+  course_id integer NOT NULL REFERENCES courses(course_id) ON DELETE CASCADE,
+  track_id integer NOT NULL REFERENCES tracks(track_id) ON DELETE CASCADE,
+  timezone_id integer NOT NULL REFERENCES timezones(tzone_id) ON DELETE CASCADE,
+  about_me text,
+  last_active timestamp,
+  created_at timestamp NOT NULL DEFAULT NOW(),
+  updated_at timestamp,
+  PRIMARY KEY (user_id)
 );
 
--- TODO: ADD REMAINING TABLES
+CREATE TABLE needs (
```

Comment:
Maybe rename `preferences`? Wondering if that's more clear.

Reply:
I think I also agree. We can maybe discuss this?

Reply:
Down to chat, but also happy with whatever you prefer. This was a bit nit picky!

Reply:
`needs` works great :)